graylog2-server/src/test/java/org/graylog2/lookup/AllowedAuxiliaryPathCheckerTest.java

```diff
@@ -22,36 +22,76 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.Collections;
-import java.util.Set;
-import java.util.SortedSet;
 import java.util.TreeSet;
 
 import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 public class AllowedAuxiliaryPathCheckerTest {
 
-    public static final String ALLOWED_PATH = "/allowed-path";
     AllowedAuxiliaryPathChecker pathChecker;
 
     @Test
-    public void success() {
-        final SortedSet<Path> paths = new TreeSet<>(Collections.singleton(Paths.get(ALLOWED_PATH)));
-        pathChecker = new AllowedAuxiliaryPathChecker(paths);
-        assertTrue(pathChecker.isInAllowedPath(ALLOWED_PATH + "/file.csv"));
+    public void inAllowedPath() throws IOException {
+        final Path filePath = validPath();
+        final Path permittedPath = validPath();
+        when(filePath.startsWith(eq(permittedPath))).thenReturn(true);
+
+        pathChecker = new AllowedAuxiliaryPathChecker(new TreeSet<>(Collections.singleton(permittedPath)));
+        assertTrue(pathChecker.fileIsInAllowedPath(filePath));
+    }
+
+    private Path validPath() throws IOException {
+        final Path filePath = mock(Path.class);
+        when(filePath.toRealPath()).thenReturn(filePath);
```

Comment:
I don't think that's a good way to test this.
The only code that's really worth having the test for is to see if `toRealPath()` is working as expected.
If we mock this, the test isn't worth much.
And `startsWith()` shouldn't be mocked either..
I would stick with how the test was done before this change, and use actual temporary files and symlinks
for the test

Reply:
Thanks a lot for the feedback @mpfz0r. This was a good change to improve the tests a lot. I've just pushed https://github.com/Graylog2/graylog2-server/pull/10244/commits/2717d95ce5f392da860d0b843e8fe0e9c3cdab19 which uses real paths and files. Specific usage of symbolic links is not included, but there is now coverage of the operative part that resolves the link (see https://github.com/Graylog2/graylog2-server/blob/2717d95ce5f392da860d0b843e8fe0e9c3cdab19/graylog2-server/src/test/java/org/graylog2/lookup/AllowedAuxiliaryPathCheckerTest.java#L100). Let me know what you think.